samples/BoltOn.Samples.Infrastructure/Data/Repositories/GradeRepository.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Threading.Tasks;
+using BoltOn.Data.CosmosDb;
+using BoltOn.Samples.Application.Abstractions.Data;
+using BoltOn.Samples.Domain.Entities;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+
+namespace BoltOn.Samples.Infrastructure.Data.Repositories
+{
+    public class GradeRepository : BaseCosmosDbRepository<Grade, CollegeDbContext>, IGradeRepository
+    {
+        public GradeRepository(CollegeDbContext collegeDbContext) : base(collegeDbContext)
+        {
+        }
+
+        public virtual Grade GetById<TId>(TId id, object partitionKey)
+        {
+            var document = _client.ReadDocumentAsync<Grade>(GetDocumentUri(id.ToString()), new RequestOptions { PartitionKey = new PartitionKey(partitionKey) })
+                .GetAwaiter().GetResult();
+            return document.Document;
+        }
+
+        public virtual async Task<Grade> GetByIdAsync<TId>(TId id, object partitionKey)
+        {
+            var document = await _client.ReadDocumentAsync<Grade>(GetDocumentUri(id.ToString()), new RequestOptions { PartitionKey = new PartitionKey(partitionKey) });
+            return document.Document;
+        }
```

Comment:
@gokulm : these are the methods which uses partition key and tested these , i couldn't overload the IRepository methods as it has only one parameter but i need extra ones here so added new methods in IGradeRepository interface.